octodns/provider/route53.py

```diff
@@ -1349,6 +1350,10 @@ def _apply(self, plan):
             mod_type = getattr(self, '_mod_{}'.format(c.__class__.__name__))
             mods = mod_type(c, zone_id, existing_rrsets)
 
+            # Rewrite NS Record to UPSERT because it always exists.
```

Comment:
I believe this may only be true in cases where there are 4 before and after. You might try updating it to 8 and then dropping it back down to 4 and seeing what that looks like. IIRC you can actually delete the original Route53 root NS records (with no way to find out what they were 😦.) 

Our main domains run with 8, 4 from each of our two active providers:

```
otter:octodns ross$
otter:octodns ross$ dig +short NS github.com
ns1.p16.dynect.net.
ns3.p16.dynect.net.
ns-520.awsdns-01.net.
ns-421.awsdns-52.com.
ns2.p16.dynect.net.
ns-1707.awsdns-21.co.uk.
ns-1283.awsdns-32.org.
ns4.p16.dynect.net.
```

Reply:
A HostedZone must contain at least one NS record for the zone itself.
You can't delete the Root NS records in AWS. But i will make it optional to use root NS per provider.

Reply:
What would happen here if you started with the default 4, added 4 more with octoDNS, and then dropped things back down to just 4. I don't believe this would allow you to delete the added NS records. 



> > `A HostedZone must contain at least one NS record for the zone itself.`

This reads like you can delete all but 1. Seems like deletes are possible, but it won't let you delete the "last" one?